### ISSUE TITLE
DownloadBuildArtifacts extractTars code is different from the one used by ExtractFiles

### ISSUE THREAD

**Issue by luiszun** [microsoft/azure-pipelines-tasks#16222]
## Note  
Issues in this repo are for tracking bugs, feature requests and questions for the tasks in this repo 

For a list:  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

If you have an issue or request for the Azure Pipelines service, use developer community instead:  

https://developercommunity.visualstudio.com/spaces/21/index.html )

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: DownloadBuildArtifacts

list here (V# not needed):  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

## Environment
- Server - Azure Pipelines
    
    - If using TFS on-premises, provide the version: 
    
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: mscatapult, buildId:806464

- Agent - Hosted 
    
    - If using Hosted agent, provide agent queue name: ubuntu 20.04

## Issue Description

Specifically this call from DownloadBuildArtifacts:
https://github.com/microsoft/azure-pipelines-tasks/blob/b2e586325791a8c52133b947692d062f72b8f492/Tasks/DownloadBuildArtifactsV0/file_helper.ts#L84

vs 

https://github.com/microsoft/azure-pipelines-tasks/blob/acc64cc7292c98597908325e53af9f898a896189/Tasks/ExtractFilesV1/extractfilestask.ts#L230

Both achieve the same, yet are different. The DownloadBuildArtifacts contains a ShellJS call to cp.
in my example, we happen to use extracted_tars (the same path hardcoded in the pipelines tasks). Causing a copy at the end of files to their same destination. This triggers a bug on ShellJS that wipes out all the files contents. I reported that (which is a regression) here: https://github.com/shelljs/shelljs/issues/1092 

Having the same codepath on this repo would  prevent people from hitting this issue in the future (or it would have been caught earlier)

In summary: DownloadBuildArtifacts with extractTars enabled is causing us to get a bunch of files without content; whereas DownloadBuildArtifacts, followed by an ExtractFiles task works fine.

### Task logs

[Enable debug logging and please provide the zip file containing all the logs for a speedy resolution]

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

### Error logs

[Insert error from the logs here for a quick overview]

**Comment by luiszun** [comment]
Closing as posted new one https://github.com/microsoft/azure-pipelines-tasks/issues/16272


The auto thing run failed, trying luck again there
